src/containers/AdminIncomingMessageList/components/IncomingMessageFilter.tsx

```diff
@@ -0,0 +1,428 @@
+import Autocomplete, {
+  AutocompleteChangeReason
+} from "@material-ui/lab/Autocomplete";
+import { css, StyleSheet } from "aphrodite";
+import React, { useState } from "react";
+
+import { dataSourceItem, DataSourceItemType } from "../../../components/utils";
+import { nameComponents } from "../../../lib/attributes";
+import { ALL_TEXTERS, UNASSIGNED_TEXTER } from "../../../lib/constants";
+
+const styles = StyleSheet.create({
+  fullWidth: {
+    width: "100%"
+  }
+});
+
+export const MESSAGE_STATUSES = {
+  all: {
+    name: "All",
+    children: ["needsResponse", "needsMessage", "convo", "messaged"]
+  },
+  needsResponse: {
+    name: "Needs Texter Response",
+    children: []
+  },
+  needsMessage: {
+    name: "Needs First Message",
+    children: []
+  },
+  convo: {
+    name: "Active Conversation",
+    children: []
+  },
+  messaged: {
+    name: "First Message Sent",
+    children: []
+  },
+  closed: {
+    name: "Closed",
+    children: []
+  }
+};
+
+export const CAMPAIGN_TYPE_FILTERS = [{ id: -1, title: "All Campaigns" }];
+
+export const TEXTER_FILTERS = [
+  { id: UNASSIGNED_TEXTER, name: "Unassigned" },
+  { id: ALL_TEXTERS, name: "All Texters" }
+];
+
+const IDLE_KEY_TIME = 500;
+
+type Tag = {
+  id: string;
+  title: string;
+};
+
+type SearchByContactName = {
+  firstName?: string | undefined;
+  lastName?: string | undefined;
+  cellNumber?: string | undefined;
+};
+
+interface IncomingMessageFilterProps {
+  isTexterFilterable: boolean;
+  isIncludeEscalatedFilterable: boolean;
+  includeEscalated: boolean;
+  includeNotOptedOutConversations: boolean;
+  includeOptedOutConversations: boolean;
+  includeArchivedCampaigns: boolean;
+  includeActiveCampaigns: boolean;
+  textersLoadedFraction: number;
+  onCampaignChanged(campaignId: number): void;
+  onTagsChanged(): void;
+  onTexterChanged(texterId: number): void;
+  onIncludeEscalatedChanged(): void;
+  onActiveCampaignsToggled(): void;
+  onArchivedCampaignsToggled(): void;
+  onNotOptedOutConversationsToggled(): void;
+  onOptedOutConversationsToggled(): void;
+  onMessageFilterChanged(messageStatuesString: string): void;
+  searchByContactName(contactDetails: SearchByContactName): void;
+  campaigns: Array<any>;
+  texters: Array<any>;
+  assignmentsFilter: { texterId: number };
+  tags: Array<Tag>;
+  tagsFilter: Array<string>;
+}
+
+const IncomingMessageFilter: React.FC<IncomingMessageFilterProps> = (props) => {
+  const [firstName, setFirstName] = useState<string>();
+  const [lastName, setLastName] = useState<string>();
+  const [cellNumber, setCellNumber] = useState<string>();
+  const [messageFilter, setMessageFilter] = useState<Array<any>>(["all"]);
+  const [showSection, setShowSection] = useState<boolean>(false);
+  const [timeoutId, setTimeoutId] = useState<any>();
+
+  const onMessageFilterSelectChanged = (
+    event: React.ChangeEvent<{ value: unknown }>,
+    _child: React.ReactNode
+  ) => {
+    const values = event.target.value as string[];
```

Comment:
This doesn't work as this is the required signature by the onChange function. If you change the type here, it will complain about incompatible types. 

<img width="830" alt="Screenshot 2022-05-10 at 12 02 25 PM" src="https://user-images.githubusercontent.com/367605/167563102-f8946c58-4b3b-459f-920c-34dab755405d.png">

Reply:
Yeah, this is the officially documented way to use it as well.